protocols/mdns/src/service.rs

```diff
@@ -213,6 +206,36 @@ impl $service_name {
     // resolves, not forcing self-referential structures on the caller.
     pub async fn next(mut self) -> (Self, MdnsPacket) {
         loop {
+            while let Some(event) = self.if_watch.next().now_or_never() {
+                let multicast = From::from([224, 0, 0, 251]);
+                let socket = self.socket.get_ref();
+                match event {
+                    Ok(IfEvent::Up(inet)) => {
+                        if inet.addr().is_loopback() {
+                            continue;
+                        }
+                        if let IpAddr::V4(addr) = inet.addr() {
+                            log::trace!("joining multicast on iface {}", addr);
+                            if let Err(err) = socket.join_multicast_v4(&multicast, &addr) {
+                                log::error!("join multicast failed: {}", err);
+                            }
+                        }
+                    }
+                    Ok(IfEvent::Down(inet)) => {
+                        if inet.addr().is_loopback() {
+                            continue;
+                        }
+                        if let IpAddr::V4(addr) = inet.addr() {
+                            log::trace!("leaving multicast on iface {}", addr);
+                            if let Err(err) = socket.leave_multicast_v4(&multicast, &addr) {
```

Comment:
As mentioned in an earlier comment, I think it would be good to get clarity on whether doing this on an interface that is down makes any sense. If this is just a no-op in this situation, there is no reason to do it and if it is an error it causes unnecessary noise in the logs. If you're sure that doing this on an interface that is down is useful and only errors when there is a serious problem other than the interface being down, I'm fine with it.

Reply:
Well it doesn't cause an error on linux or windows. I'll try to investigate more deeply especially on linux, but we are supporting any platform that rust compiles to I guess.

Reply:
> I'll try to investigate more deeply especially on linux, [..]

Just curious, did you come to a conclusion on the matter?